internal/controller/standalone_pgadmin/statefulset.go

```diff
@@ -33,6 +35,27 @@ func (r *PGAdminReconciler) reconcilePGAdminStatefulSet(
 	ctx context.Context, pgadmin *v1beta1.PGAdmin,
 	configmap *corev1.ConfigMap, dataVolume *corev1.PersistentVolumeClaim,
 ) error {
+	// In a previous version of PGO, the PGAdmin StatefulSet was not assigned a Parallel Pod Management Policy
+	// which meant that in certain scenarios, a failed Pod could not recover (e.g. a bad image) and would have
+	// to be manually deleted. We now assign the Parallel Pod Management Policy, however, this is not a property
+	// that can be patched, so if we run into a StatefulSet from the previous PGO version that does not have the
+	// parallel policy in place, we must delete and recreate the StatefulSet.
+	existing := &appsv1.StatefulSet{ObjectMeta: naming.StandalonePGAdmin(pgadmin)}
+	if err := errors.WithStack(r.Client.Get(ctx, client.ObjectKeyFromObject(existing), existing)); err != nil {
+		if !apierrors.IsNotFound(err) {
+			return err
+		}
+	} else {
+		if existing.Spec.PodManagementPolicy != appsv1.ParallelPodManagement {
+			// We want to delete the STS without affecting the Pods, so we set the PropagationPolicy to Orphan.
+			// The orphaned Pods will be claimed by the new StatefulSet that gets created below.
+			propagate := client.PropagationPolicy(metav1.DeletePropagationOrphan)
+			if err = errors.WithStack(r.Client.Delete(ctx, existing, propagate)); err != nil {
```

Comment:
🔧 There is a chance that the object existed when we did the Get (read) and has since gone away when we Delete (write). When we want the object gone, Not Found is okay, so we can use `client.IgnoreNotFound()`.

🔧 There is a chance that the object changed after we did the Get (read). When deleting an object because of something you saw on it, it is good to use preconditions to delete the object you saw rather than the one that might be there now. In the PostgresCluster reconciler, it looks like we used `deleteControlled` most of the time for this.

```go
		uid := existing.GetUID()
		version := existing.GetResourceVersion()
		exactly := client.Preconditions{UID: &uid, ResourceVersion: &version}
		propagate := client.PropagationPolicy(metav1.DeletePropagationOrphan)

		return errors.WithStack(
			client.IgnoreNotFound(r.Client.Delete(ctx, existing, exactly, propagate))
```